### PR TITLE
SNOW-1734018 SPCS command fixes

### PR DESCRIPTION
**Pull request by sfc-gh-sichen**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description

SNOW-1734018

Updates the following `snow spcs` commands:
Deprecations
* Added deprecation warning in the description of `snow spcs service status` and `snow spcs image-repository list-tags`.

New additions
* Added `snow spcs service list-instances`, `snow spcs service list-containers` and `snow spcs service list-roles` commands, which support fetching information about all instances/containers/service roles in a service.

Fixes and improvements
* `snow spcs service set` now supports `--eai-name` to update external access integrations for a service.
* `snow spcs service list-images` now displays image tag and digest.

### CODE REVIEW

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -18,11 +18,13 @@
...
 
 ## Deprecations
+* Added deprecation warning in the description of `snow spcs service status` and `snow spcs image-repository list-tags`.
 
 ## New additions
-
+* Added `snow spcs service list-instances`, `snow spcs service list-containers` and `snow spcs service list-roles` commands, which support fetching information about all instances/containers/service roles in a service.
 
 ## Fixes and improvements
+* `snow spcs service set` now supports `--eai-name` to update external access integrations for a service.
```
Comment (sfc-gh-yiwang) [inline RELEASE-NOTES.md:27]: This also looks like a new addition.

Comment (sfc-gh-yiwang) [inline RELEASE-NOTES.md:27]: Probably also mentioning that we improve the `snow spcs service list-images` result by including enriched info like tag and sha

File: tests/spcs/test_image_repository.py [hunk 2]
```diff
@@ -140,43 +140,47 @@ def test_create_repository_already_exists(mock_handle, mock_execute):
...
-    mock_get_images.return_value.text = '{"repositories":["baserepo/super-cool-repo"]}'
+    mock_list_images.return_value = cursor
 
     result = runner.invoke(
         ["spcs", "image-repository", "list-images", "IMAGES", "--format", "JSON"]
     )
 
     assert result.exit_code == 0, result.output
-    assert json.loads(result.output) == [{"image": "/DB/SCHEMA/IMAGES/super-cool-repo"}]
+    assert "/db/schema/repo/echo_service" in result.output
```
Comment (sfc-gh-yiwang) [inline tests/spcs/test_image_repository.py:170]: nit: why don't we assert cursor == result directly?

File: tests/spcs/test_services.py [hunk 3]
```diff
@@ -513,6 +513,96 @@ def test_list_endpoints_cli(mock_list_endpoints, mock_cursor, runner):
...
+            "instance_id",
+            "status",
+        ],
+    )
+    mock_list_instances.return_value = cursor
+    result = runner.invoke(["spcs", "service", "list-instances", service_name])
+
+    mock_list_instances.assert_called_once_with(service_name=service_name)
+    assert result.exit_code == 0
+    assert "TEST_SERVICE" in result.output, str(result.output)
```
Comment (sfc-gh-yiwang) [inline tests/spcs/test_services.py:545]: nit: why don't we assert cursor == result directly?

**Review by sfc-gh-yiwang — DISMISSED** [review]
LGTM. Please ensure the SPCS integration tests runs successfully before merging.

File: tests/spcs/test_image_repository.py (hunk 2 above)
Comment (sfc-gh-sichen) [reply tests/spcs/test_image_repository.py:170]: The output is in table format. I guess we could make it json and compare. But the return value is faked entirely. For example, it has the same effect as making `cursor` return `foo` and assert the result is the same as `foo`.

File: RELEASE-NOTES.md (hunk 1 above)
Comment (sfc-gh-sichen) [reply RELEASE-NOTES.md:27]: Good point! I will update.

**Review by sfc-gh-astus — APPROVED** [review]

**Comment by sfc-gh-sichen** [comment]
Integration test failed but doesn't look like related
FAILED tests_integration/test_temporary_connection.py::test_temporary_connection - ValueError: Invalid private key

File: tests_integration/spcs/spec/job_service_spec.yaml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 spec:
   containers:
   - name: main
-    image: /snowcli_db/public/snowcli_repository/test_counter
+    image: /sichen_db/test_schema/sichen_repo/test_counter
```
[the remark this answers is not in the document]
Comment (sfc-gh-astus) [reply tests_integration/spcs/spec/job_service_spec.yaml:4]: @sfc-gh-sichen did you update images? If yes, please update scripts too https://github.com/snowflakedb/snowflake-cli/tree/main/tests_integration/spcs/docker .

[the remark this answers is not in the document]
Comment (sfc-gh-sichen) [reply tests_integration/spcs/spec/job_service_spec.yaml]: sorry reverted

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**